### PR TITLE
Fix DG issues

### LINKED ISSUE #175: DG: changes to task manager UML class

### PR DESCRIPTION
**Pull request by clarencepohh**
Closes #175, #179, #187, #188.

Changes in organization (#188): 
- Deleted redundant logging and error handling segments
- Moved DateUtils section up 1 level
- Moved Task Type segment to TaskManager Component segment
- Moved Retrieving tasks from the TaskManager, Updating a Task, Adding Tasks, Deleting Tasks and Interfacing with Storage class to TaskManager Component segment
- Shifted Target User Profile and Value proposition under Product Scope segment
- Remade Table of Contents

Manual Testing Segment Linking to UG (#187): Added section for UG link

#182 Exceptions: Added some exceptions. Will not close this issue yet however, for all to check

Review TaskManager Component: How it Works (#179): Fixed typo 

Changes to TaskManager UML Class Diagram (#175): Removed exceptions from diagram. Add association arrow from TaskManager to Task class. 

Sequence Diagram for Data (#176): To be done in a future PR (tomorrow probably)

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1,37 +1,29 @@
...
-  * [Target user profile](#target-user-profile)
-  * [Value proposition](#value-proposition)
+    * [Target user profile](#target-user-profile)
+    * [Value proposition](#value-proposition)
   * [User Stories](#user-stories)
   * [Non-Functional Requirements](#non-functional-requirements)
-* [Glossary](#glossary)
-* [Instructions for manual testing](#instructions-for-manual-testing)
+  * [Glossary](#glossary)
+  * [Instructions for manual testing](#instructions-for-manual-testing)
```
Comment (NicholasTanYY) [inline docs/DeveloperGuide.md:26]: Thanks for fixing the formatting of the headers and the table of contents!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -213,34 +205,29 @@ while (true) {
...
 
-## Retrieving tasks from the TaskManager
-### getTasksForDate Method
+#### Task Types
+`TaskManager` supports different task types to be created by the user.
+The `TaskType` enum defines the different types of tasks supported by the application: `TODO`, `EVENT`, and `DEADLINE`.
+There is also a `INVALID` type, used for error handling.
+
+### Retrieving tasks from the TaskManager
+#### getTasksForDate Method
```
Comment (NicholasTanYY) [inline docs/DeveloperGuide.md:230]: Good relocation of the description for TaskTypes!

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -303,47 +289,47 @@ public void updateManager(Scanner scanner, WeekView weekView, MonthView monthVie
...
 
 The `addTask` method orchestrates the creation and addition of a new task to the TaskManager. It accepts parameters including the task date, description, type, and relevant dates/times for events and deadlines. Depending on the task type, it constructs the corresponding task object (Todo, Event, or Deadline) and integrates it into the tasks map.
 
 #### Method Signature
 
 ```
-public static void addTask(LocalDate date, String taskDescription, TaskType taskType, String[] dates,
-                           String[] times) throws TaskManagerException
+public static void addTask(LocalDate date, String taskDescription, TaskType taskType,
+        String[] dates, String[] times) throws TaskManagerException {
```
Comment (NicholasTanYY) [inline docs/DeveloperGuide.md:332]: Good spot on updating the parameters for the method signature!

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -377,55 +367,62 @@ public void addManager(Scanner scanner, WeekView weekView, MonthView monthView,
 - `taskTypeString`: String representing the task type.
 - `taskDescription`: Description of the task.
 
-### Handling User Input
+#### Exceptions
+
+- `TaskManagerException`: Thrown if there is an error in managing tasks.
+- `DateTimeParseException`: Thrown if there is an error parsing the date or time.
+
+#### Handling User Input
```
Comment (NicholasTanYY) [inline docs/DeveloperGuide.md:375]: Great job on adding these exceptions to make it consistent with the rest of the DG!

File: docs/diagrams/class/Data.puml [hunk 5]
```diff
@@ -87,4 +90,6 @@ LOW
 
 hide circle
 
+Task "*"  <- "1"TaskManager
+
```
Comment (NicholasTanYY) [inline docs/diagrams/class/Data.puml:94]: Good update on the UML diagram for TaskManager to reflect the latest changes!

**Review by NicholasTanYY — APPROVED** [review]
Overall, great job on updating the DG to increase coherence and making other changes! LGTM

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -1,37 +1,29 @@
...
-    * [Task Types](#task-types)
-    * [Error Handling <br>](#error-handling-br)
+  * [Utilities](#utilities)
   * [TaskManager Component](#taskmanager-component)
-  * [Retrieving tasks from the TaskManager](#retrieving-tasks-from-the-taskmanager)
-  * [Updating a Task](#updating-a-task)
-  * [Adding Tasks](#adding-tasks)
-  * [Deleting Tasks](#deleting-tasks)
-  * [Interfacing with Storage class](#interfacing-with-storage-class)
+    * [Retrieving tasks from the TaskManager](#retrieving-tasks-from-the-taskmanager)
```
Comment (Mahesh1772) [inline docs/DeveloperGuide.md:10]: Good job fixing the indentation.

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -303,47 +289,47 @@ public void updateManager(Scanner scanner, WeekView weekView, MonthView monthVie
...
+#### `addTask` Method
 
 The `addTask` method orchestrates the creation and addition of a new task to the TaskManager. It accepts parameters including the task date, description, type, and relevant dates/times for events and deadlines. Depending on the task type, it constructs the corresponding task object (Todo, Event, or Deadline) and integrates it into the tasks map.
 
 #### Method Signature
 
 ```
-public static void addTask(LocalDate date, String taskDescription, TaskType taskType, String[] dates,
-                           String[] times) throws TaskManagerException
+public static void addTask(LocalDate date, String taskDescription, TaskType taskType,
```
Comment (Mahesh1772) [inline docs/DeveloperGuide.md:331]: Good job on adhering to a standard indentation

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -489,7 +489,7 @@ The `loadTasksFromFile` method reads the tasks from the file and populates the `
 
 ### Exceptions
 
-![Exceptions.png](images%2Fclass%2FExceptions.png)
+![Exceptions.png](images/class/Exceptions.png)
```
Comment (Mahesh1772) [inline docs/DeveloperGuide.md:492]: Thanks for updating the image!

File: docs/diagrams/class/Data.puml [hunk 9]
```diff
@@ -5,22 +5,32 @@ skinparam classAttributeIconSize 0
...
 + markTaskAsNotCompleted (date: LocalDate, taskIndex: int)
 + getTasksForDate (date: LocalDate)
 + addManager (scanner: Scanner, weekView: WeekView, monthView: MonthView, inMonthView: boolean, \n                      action: String, day: String, taskTypeString: String, taskDescription: String)
++ freeTimeManager (weekView: WeekView, monthView: MonthView, inMonthView: boolean, day: String)
 + markManager (weekView: WeekView, monthView: MonthView, inMonthView: boolean, day: String, taskIndex: int)
 + handleMarkingOfTask (taskIndex: int, date: LocalDate)
 + priorityManager (weekView: WeekView, monthView: MonthView, inMonthView: boolean, day: String, taskIndex: int, \n                          priorityLevelString: String)
 + updateManager (scanner: Scanner, weekView: WeekView, monthView: MonthView, inMonthView: boolean, \n                           taskManager: TaskManager, day: int, taskIndex: int, newDescription: String)
 + addTasksFromFile (tasksFromFile: Map<LocalDate, List<Task>>)
++ getFreeTimeSlots (events: List<Task>, currentDate: LocalDate)
```
Comment (Mahesh1772) [inline docs/diagrams/class/Data.puml:29]: Good job editing the Data diagram

File: docs/images/class/Data.jpg
Comment (Mahesh1772) [inline docs/images/class/Data.jpg:1]: Looks way more organize!

**Review by Mahesh1772 — COMMENTED** [review]
LTGM! Nice job on DG

**Pull request merged**